src/site/resources/styleguides/google-java-style-20180523/javaguide.html

```diff
@@ -367,13 +380,20 @@ <h3 id="s4.4-column-limit">4.4 Column limit: 100</h3>
   3.2 <a href="#s3.2-package-statement">Package statement</a> and
   3.3 <a href="#s3.3-import-statements">Import statements</a>).</li>
 
-  <li>Command lines in a comment that may be cut-and-pasted into a shell.</li>
+  <li>Command lines in a comment that may be copied-and-pasted into a shell.</li>
+
+  <li>Very long identifiers, on the rare occasions they are called for, are allowed to exceed the
+  column limit. In that case, the valid wrapping for the surrounding code is as produced by
```

Comment:
#### Checkstyle.java
```java
package com.examples.columnlimit;

/** testing. */
public class Checkstyle {
  int aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa = 99;

  public void testtttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt() {
    System.out.println("heloooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo");
  }
}
```

output:
```
$ java -jar ..\checkstyle-10.17.0-all.jar -c ..\google_checks.xml .\Checkstyle.java
Starting audit...
[WARN] \Checkstyle.java:5: Line is longer than 100 characters (found 109). [LineLength]
[WARN] \Checkstyle.java:7: Line is longer than 100 characters (found 106). [LineLength]
[WARN] \Checkstyle.java:8: Line is longer than 100 characters (found 110). [LineLength]
Audit done.
```

Formatted code by google-java-format:
```java
package com.examples.columnlimit;

/** testing. */
public class Checkstyle {
  int
      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa =
          99;

  public void
      testtttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt() {
    System.out.println(
        "heloooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo");
  }
}
```

The formatter just formats the code to make it under column width 100. 

If we have some code we can't wrap to make it under column 100, this can lead to a problem. User will probably need to use suppression for it.  How should we approach this?

Reply:
> The formatter just formats the code to make it under column width 100.

It is not true when you have an identifier of more than 100 characters. It is ok that formatter wraps it. On the Checkstyle side we may probably need to revisit the LineLength check and extend it with an option to skip long identifiers.

@Zopsss, please create an issue to handle this case; let's move the discussion there. Reply to this comment with the link to the created issue.

Reply:
Opened issue regarding this: https://github.com/checkstyle/checkstyle/issues/15233

Reply:
Placed in todo queue.

Reply:
@Zopsss, I do not see PR with update on coverage table for 4.4 to reference https://github.com/checkstyle/checkstyle/issues/15233 - please do.

Reply:
done https://github.com/checkstyle/checkstyle/pull/15426

Reply:
review thread is resolved